File: Backend/controller/kas/kip17/kip17TokenMinting.js
const kip17 = require('../../../service/kas/kip17');

const kip17TokenMinting = async function (req, res) {
    try {
        const tokenMinting = await kip17.kasKip17TokenMinting(req.body.category, req.body.to, req.body.token_id, req.body.metadata_uri);

        if (tokenMinting['statusCode'] == 200) {
            res.status(200);
            res.json({statusCode: 200});
        } else { 
            const errorCode = tokenMinting['data']['code'];
            res.status(tokenMinting['statusCode']);

            switch (errorCode) {
                case 1000000:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "카테고리가 올바르지 않습니다."});
                    break;
                case 1100251:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "컨트랙트 배포를 확인중입니다."});
                    break;
                case 1104700:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "유효하지 않은 별칭입니다."});
                    break;
                case 1010009:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "자격증명 정보가 유효하지 않습니다."});
                    break;
                case 1100050:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "토큰 소유자 주소가 잘못 되었습니다."});
                    break;
                case 1104400:
                    res.json({statusCode: tokenMinting['statusCode'], msg: "중복된 별칭입니다."});
                    break;
                default:
                    res.json({statusCode: tokenMinting['statusCode'], msg: tokenMinting['data']['message']});
                    break;
            }
        }
    } catch (e) {
        console.error(e);
    }
}

module.exports = kip17TokenMinting;

File: Etiquette/lib/Providers/KAS/Kip17/kip17_token_minting.dart
import 'package:dio/dio.dart';
import 'package:Etiquette/Models/dio_models.dart';
import 'package:Etiquette/Models/serverset.dart';

// kip17 토큰 발행
Future<Map<String, dynamic>> kip17TokenMinting(String category, String to, String token_id, String metadata_uri) async {
  Dio dio = getBasicDio();
  String url = "$SERVER_IP/kas/kip17/tokenMinting";
  try {
    final res = await dio.post(url, data: {
      "category": category,
      "to": to,
      "token_id": token_id,
      "metadata_uri": metadata_uri,
    });
    Map<String, dynamic> data = res.data;
    return data;
  } on DioError catch (e) {
    final handleError = e.response?.data;
    if (handleError == null) {
      Map<String, dynamic> data = {
        "statusCode": 400,
        "msg": "서버와의 연결이 원활하지 않습니다.",
      };
      return data;
    } else {
      Map<String, dynamic> data = {
        "statusCode": e.response?.statusCode,
        "msg": handleError["msg"],
      };
      return data;
    }
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}